lib/generator/command_line.rb

```diff
@@ -1,26 +1,21 @@
 require 'logger'
-require 'optparse'
 
 module Generator
   # Processes the command line arguments and sets everthing up and returns a
   # generator that can be called
   class CommandLine
     def initialize(paths)
       @paths = paths
-      @options = default_options
     end
 
     def parse(args)
-      parse_options(args)
-      generator if valid?
+      parser = GeneratorOptparser.new(args, @paths)
+      @options = parser.parse_options
+      generator if parser.valid?
```

Comment:
very minor: We want to know if the options are valid, not if the parser is valid, so this reads oddly to me.